src/resolvers/mutations/CreateUser.ts

```diff
@@ -19,12 +22,45 @@ export const CreateUser = (_, { user: CreateUserInput }, { request, response }):
     }
     if (!payload.userId) {
       response.statusCode = HttpStatusCode.BAD_REQUEST;
-      throw Error('Malformed token payload');
+      throw new DetailedError('Malformed token payload');
     }
 
-    return { id: 1, name: 'name', email: 'email', birthDate: new Date(), cpf: '10020030012', password: ''};
+    let errors: DetailedError[] = verifyPassword(user.password);
+    const isEmailUnique = !(await db.manager.findOne(User, { email: user.email }));
+    if (!isEmailUnique) {
+      errors.push(new DetailedError('Email already used'));
+    }
+    
+    if (errors.length > 0) {
+      throw new ErrorPack('Validation errors', errors);
+    }
+
+    const salt = bcryptjs.genSaltSync(10);
```

Comment:
O salt é o mesmo em todas as execuções? Se ele for diferente a comparação vai dar ruim mesmo se a senha estiver correta, né?

Reply:
Uma das estratégias que usamos para salt é ter um salt do sistema, que vem de uma variável de ambiente. Assim ela tem sempre o mesmo valor (caso não seja trocada posteriormente)

Reply:
O salt é diferente em cada execução, mas ele é concatenado com o hash final. Então o `bcryptjs.compare` consegue deduzir o salt e descobrir o password corretamente.
[fonte](https://auth0.com/blog/hashing-in-action-understanding-bcrypt/)